embassy-nrf/src/ppi/mod.rs

```diff
@@ -137,9 +137,9 @@ impl Task {
         Self(ptr)
     }
 
-    // Triggers this task.
+    /// Triggers this task.
     pub unsafe fn trigger(&mut self) {
-        self.0.write(|w| unsafe { w.bits(1) });
+        *self.0.as_ptr() = 1;
```

Comment:
register access needs to be volatile reads/writes, not regular pointer reads/writes. (needed to prevent the compiler from reoderding / optimizing them out).